hooks/pre-command

```diff
@@ -18,11 +18,54 @@ ecr_exists() {
     --query 'repositories[0].registryId'
 }
 
+get_ecr_arn() {
+  local repository_name="${1}"
+  aws ecr describe-repositories \
+    --repository-names "${repository_name}" \
+    --output text \
+    --query 'repositories[0].repositoryArn'
+}
+
+get_current_ecr_tags() {
+  local resource_arn="${1}"
+  aws ecr list-tags-for-resource \
+    --resource-arn "${resource_arn}" \
+    --output text \
+    --query 'tags[*].Key'
+}
+
+get_ecr_tags() {
+  local result=""
+  env
+  while IFS='=' read -r name _ ; do
+    if [[ $name =~ ^(BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_ECR_TAGS_) ]] ; then
+      # Handle plain key=value, e.g
+      # ecr-tags:
+      #   KEY_NAME: 'key-value'
+      echo "-------"
+      echo $name
+      key_name=$(echo "${name}" | sed 's/^BUILDKITE_PLUGIN_DOCKER_ECR_CACHE_ECR_TAGS_//')
+      key_value="${!name}"
+      key_value_str="Key=${key_name},Value=${key_value}"
+      result="${result} ${key_value_str}"
+    fi
+  done < <(env | sort)
+
+  echo $result
+}
+
 upsert_ecr() {
   local repository_name="${1}"
   local max_age_days="${2}"
+  local ecr_tags="$(get_ecr_tags)"
+
   if ! ecr_exists "${repository_name}"; then
-    aws ecr create-repository --repository-name "${repository_name}"
+    aws ecr create-repository --repository-name "${repository_name}" --tags "${ecr_tags}"
+  else
+    local ecr_arn="$(get_ecr_arn "${repository_name}")"
+    local current_ecr_tags="$(get_current_ecr_tags "${ecr_arn}")"
+    aws ecr untag-resource --resource-arn ${ecr_arn} --tag-keys ${current_ecr_tags}
```

Comment:
IMO don't untag anything, just apply the tags you have. Gets too complicated otherwise, and may start to impact manual tags